docs/user_guide/header-links.rst

```diff
@@ -0,0 +1,243 @@
+For example, to change the number of displayed header links to be ``4`` instead of ``5``:abbr:
+
+.. code-block:: python
+
+   html_theme_options = {
+     "header_links_before_dropdown": 4
+   }
+
+.. _icon-links:
+
+Icon links
+==========
+
+Icon links are a collection of images and icons that each link to a page or external site.
+They are helpful if you wish to display social media icons, GitHub badges, or project logos.
+
+These links take the following form:
+
+.. code:: python
+
+   html_theme_options = {
+       ...
+       "icon_links": [
+           {
+               # Label for this link
+               "name": "GitHub",
+               # URL where the link will redirect
+               "url": "https://github.com/<your-org>/<your-repo>",  # required
+               # Icon class (if "type": "fontawesome"), or path to local image (if "type": "local")
+               "icon": "fab fa-github-square",
+               # The type of image to be used (see below for details)
+               "type": "fontawesome",
+           }
+      ]
+   }
+
+There are two kinds of icons you can use, described below:
+
+FontAwesome icons
+-----------------
+
+`FontAwesome <https://fontawesome.com/>`_ is a collection of icons that are
+commonly used in websites. They include both generic shape icons (e.g., "arrow-down"),
+as well as brand-specific icons (e.g. "github").
+
+You can use FontAwesome icons by specifying ``"type": "fontawesome"``, and
+specifying a FontAwesome class in the ``icon`` value.
+The value of ``icon`` can be any full
+`FontAwesome 5 Free <https://fontawesome.com/icons?d=gallery&m=free>`__ icon.
+In addition to the main icon class, e.g. ``fa-cat``, the "style" class must
+also be provided e.g. `fab` for *branding*, or `fas` for *solid*.
+
+Here are several examples:
+
+.. code:: python
+
+   html_theme_options = {
+       ...
+       "icon_links": [
+           {
+               "name": "GitHub",
+               "url": "https://github.com/<your-org>/<your-repo>",
+               "icon": "fab fa-github-square",
+               "type": "fontawesome",
+           },
+           {
+               "name": "GitLab",
+               "url": "https://gitlab.com/<your-org>/<your-repo>",
+               "icon": "fab fa-gitlab",
+               "type": "fontawesome",
+           },
+           {
+               "name": "Twitter",
+               "url": "https://twitter.com/<your-handle>",
+               "icon": "fab fa-twitter-square",
+               # The default for `type` is `fontawesome` so it is not actually required in any of the above examples as it is shown here
+           },
+           {
+               "name": "Mastodon",
+               "url": "https://<your-host>@<your-handle>",
+               "icon": "fab fa-mastodon",
+               "attributes": {
+                  "target" : "_blank",
+                  "rel" : "noopener me",
+               }
```

Comment:
Suggest cutting this here, because it hasn't been explained yet, and there's a more complete example/explanation later 
```suggestion
```